### PR TITLE
Fix an issue that Android libraries was not published

### PR DESCRIPTION
**Pull request by kubode**
Fixes: https://github.com/mockk/mockk/issues/883

The official Android documentation about publications is [here](https://developer.android.com/studio/publish-library/upload-library#create-pub).

I confirmed that `mockk-android` is published with the following command.

```
$ ./gradlew publishToMavenLocal
$ tree ~/.m2/repository/io/mockk/
~/.m2/repository/io/mockk
├── mockk
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-1.12.7-SNAPSHOT-kotlin-tooling-metadata.json
│   │   ├── mockk-1.12.7-SNAPSHOT-sources.jar
│   │   ├── mockk-1.12.7-SNAPSHOT.jar
│   │   ├── mockk-1.12.7-SNAPSHOT.module
│   │   └── mockk-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-agent
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-agent-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-agent-1.12.7-SNAPSHOT-kotlin-tooling-metadata.json
│   │   ├── mockk-agent-1.12.7-SNAPSHOT-sources.jar
│   │   ├── mockk-agent-1.12.7-SNAPSHOT.jar
│   │   ├── mockk-agent-1.12.7-SNAPSHOT.module
│   │   └── mockk-agent-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-agent-android
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-agent-android-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-agent-android-1.12.7-SNAPSHOT.aar
│   │   ├── mockk-agent-android-1.12.7-SNAPSHOT.module
│   │   └── mockk-agent-android-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-agent-api
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-agent-api-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-agent-api-1.12.7-SNAPSHOT-kotlin-tooling-metadata.json
│   │   ├── mockk-agent-api-1.12.7-SNAPSHOT-sources.jar
│   │   ├── mockk-agent-api-1.12.7-SNAPSHOT.jar
│   │   ├── mockk-agent-api-1.12.7-SNAPSHOT.module
│   │   └── mockk-agent-api-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-agent-api-jvm
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-agent-api-jvm-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-agent-api-jvm-1.12.7-SNAPSHOT-sources.jar
│   │   ├── mockk-agent-api-jvm-1.12.7-SNAPSHOT.jar
│   │   ├── mockk-agent-api-jvm-1.12.7-SNAPSHOT.module
│   │   └── mockk-agent-api-jvm-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-agent-jvm
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-agent-jvm-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-agent-jvm-1.12.7-SNAPSHOT-sources.jar
│   │   ├── mockk-agent-jvm-1.12.7-SNAPSHOT.jar
│   │   ├── mockk-agent-jvm-1.12.7-SNAPSHOT.module
│   │   └── mockk-agent-jvm-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-android
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-android-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-android-1.12.7-SNAPSHOT.aar
│   │   ├── mockk-android-1.12.7-SNAPSHOT.module
│   │   └── mockk-android-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-dsl
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-dsl-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-dsl-1.12.7-SNAPSHOT-kotlin-tooling-metadata.json
│   │   ├── mockk-dsl-1.12.7-SNAPSHOT-sources.jar
│   │   ├── mockk-dsl-1.12.7-SNAPSHOT.jar
│   │   ├── mockk-dsl-1.12.7-SNAPSHOT.module
│   │   └── mockk-dsl-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
├── mockk-dsl-jvm
│   ├── 1.12.7-SNAPSHOT
│   │   ├── maven-metadata-local.xml
│   │   ├── mockk-dsl-jvm-1.12.7-SNAPSHOT-javadoc.jar
│   │   ├── mockk-dsl-jvm-1.12.7-SNAPSHOT-sources.jar
│   │   ├── mockk-dsl-jvm-1.12.7-SNAPSHOT.jar
│   │   ├── mockk-dsl-jvm-1.12.7-SNAPSHOT.module
│   │   └── mockk-dsl-jvm-1.12.7-SNAPSHOT.pom
│   └── maven-metadata-local.xml
└── mockk-jvm
    ├── 1.12.7-SNAPSHOT
    │   ├── maven-metadata-local.xml
    │   ├── mockk-jvm-1.12.7-SNAPSHOT-javadoc.jar
    │   ├── mockk-jvm-1.12.7-SNAPSHOT-sources.jar
    │   ├── mockk-jvm-1.12.7-SNAPSHOT.jar
    │   ├── mockk-jvm-1.12.7-SNAPSHOT.module
    │   └── mockk-jvm-1.12.7-SNAPSHOT.pom
    └── maven-metadata-local.xml

20 directories, 72 files
```

### CODE REVIEW

File: buildSrc/src/main/kotlin/buildsrc/convention/mockk-publishing.gradle.kts [hunk 1]
```diff
@@ -64,13 +64,23 @@ publishing {
...
+        }
+    }
     publications.withType<MavenPublication>().configureEach {
         createMockKPom {
             name.set(provider { mavenName })
             description.set(provider { mavenDescription })
         }
 
-        artifact(tasks.provider<Jar>("javadocJar"))
+        artifact(tasks.provider<Task>("javadocJar"))
```
Comment (kubode) [inline buildSrc/src/main/kotlin/buildsrc/convention/mockk-publishing.gradle.kts:83]: The reason for this change is that the following error occurs.

```
> The task 'javadocJar' (org.gradle.jvm.tasks.Jar) is not a subclass of the given type (org.gradle.api.tasks.bundling.Jar).
```

**Comment by Raibaz** [comment]
Brilliant, thanks a lot!

File: buildSrc/src/main/kotlin/buildsrc/convention/mockk-publishing.gradle.kts (hunk 1 above)
Comment (kubode) [reply buildSrc/src/main/kotlin/buildsrc/convention/mockk-publishing.gradle.kts]: It seems to have been caused by importing `org.gradle.jvm.tasks.Jar` in `android-library.gradle.kts`.
Removing import from `android-library.gradle.kts` solved the problem.

Comment (aSemy) [reply buildSrc/src/main/kotlin/buildsrc/convention/mockk-publishing.gradle.kts]: Nice catch, thanks

**Pull request merged**